### DISCUSSION TITLE
How to decompose queryType( t.crud) into multiple file

### DISCUSSION THREAD

**Discussion by aakashkag** [prisma/prisma#2862] [Q&A] [answered]
Currently I am using

```ts
const query = queryType({
    definition(t) {
        t.crud.organization({
            filtering: true,
            ordering: true,
            pagination: true,
        });

        t.crud.subOrganizations({
            filtering: true,
            ordering: true,
            pagination: true,
        });
}
});
```

But it will get bigger and bigger if i add some custome resolver. 
Can we decompose it?. I would like to create one  file corrosponding each  model and import at single place.
I tried to decompose it but gave error. "Error: Query was already defined and imported as a type, check the docs for extending types".

**Comment by ryands17** [comment, marked as answer]
@aakashkag Try using `extendType` as follows:

```ts
const something = extendType({
  type: 'Query',
  definition(t) {
    // your queries here  
  }
})
```

the same will work for mutations as well

**Reply by aakashkag** [reply]
thanks dear. :)

**Comment by ludralph** [comment]
Hi @aakashkag,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
